src/libcore/atomic.rs

```diff
@@ -1089,3 +1090,21 @@ pub fn fence(order: Ordering) {
         }
     }
 }
+
+macro_rules! impl_Debug {
+    ($($t:ident)*) => ($(
+        impl fmt::Debug for $t {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                write!(f, "Atomic {{ {:?} }}", self.load(Ordering::Relaxed))
```

Comment:
These implementations should match the [relevant RFC](https://github.com/rust-lang/rfcs/blob/master/text/0565-show-string-guidelines.md#debugging-fmtdebug). I'm not sure if these should be treated like smart pointers, in which case this would just be `self.load(Ordering::Relaxed).fmt(fmt)`, or a struct, in which case something like this should suffice:
```rust
fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
    f.debug_tuple(stringify!($t)).field(&self.load(Ordering::Relaxed)).finish()
}
```